src/main/java/org/apache/sling/servlets/post/exceptions/PreconditionViolatedPersistenceException.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.servlets.post.exceptions;
+
+import org.apache.sling.api.resource.PersistenceException;
+
+
+/**
+ *  Indicates that the input does not meet necessary precondition.
+ *  In that case the client should redo the request with a changed input.
+ *
+ */
+
+public class PreconditionViolatedPersistenceException extends PersistenceException {
```

Comment:
My only substantial comment is related to the exceptions. Have you considered making them unchecked, e.g. extend from `SlingException`? The advantage would be that we don't need to tweak signatures if we decide to add new exceptions.

Reply:
Well, looking at the commit history the number of changes to the SlingPostServlet in the last years is quite low, meaning the overall functionality is quite stable. I wouldn't expect that this situation changes much in the future :-)

While I understand your concern from a point of the number of changes you need to make to introduce new exception types, I prefer to extend these exceptions from the PersistenceException, because that's what they are.  And in that case any exception handler for PersistenceException is handling these subtypes as well, it's not forcing you to write new handlers.

(We could agree to include it into the next overhaul of the SlingPostServlet, but right now I don't see a big value in it; as written, I don't believe that big changes are looming here.)

Reply:
Ack, thanks.